code/game/objects/items/weapons/twohanded.dm

```diff
@@ -1000,22 +1000,30 @@
 		return
 	var/turf/new_item_loc = get_step(current_item_loc, user.dir)
 	var/obj/machinery/disposal/target_bin = locate(/obj/machinery/disposal) in new_item_loc.contents
+	var/obj/structure/janitorialcart/jani_cart = locate(/obj/structure/janitorialcart) in new_item_loc.contents
+	var/obj/vehicle/janicart/jani_vehicle = locate(/obj/vehicle/janicart) in new_item_loc.contents
 	var/trash_amount = 1
 	for(var/obj/item/garbage in current_item_loc.contents)
 		if(!garbage.anchored)
+			if(jani_vehicle?.mybag && garbage.w_class <= WEIGHT_CLASS_SMALL)
+				move_into_storage(user, jani_vehicle.mybag, garbage)
+			if(jani_cart?.mybag && garbage.w_class <= WEIGHT_CLASS_SMALL)
```

Comment:
```suggestion
			else if(jani_cart?.mybag && garbage.w_class <= WEIGHT_CLASS_SMALL)
```
Else you can get the edge case where having both a bin, jani_cart and vehicle on the same tile will insert the item into all three of them